### PR TITLE
Add prettier formatting: fix #57

### LINKED ISSUE #57: Add prettier formatting
Add prettier to force code formatting.

### PR DESCRIPTION
**Pull request by zh99998**
notice that prettier recommends use 2 space as indent, not 4.

since Google https://github.com/angular/angular-cli/blob/master/.editorconfig#L8 、 StandardJS https://standardjs.com/ recommends 2 space too. maybe it is better?

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -4,7 +4,7 @@ insert_final_newline = true
 
 [*.{js,json}]
 indent_style = space
-indent_size = 4
+indent_size = 2
```
Comment (perry-mitchell) [inline .editorconfig]: I prefer 4 spaces, and find it more readable.

File: source/auth.js [hunk 2]
```diff
@@ -1,9 +1,9 @@
 "use strict";
 
 function generateBasicAuthHeader(username, password) {
-    return "Basic " + (new Buffer(username + ":" + password)).toString("base64");
+  return "Basic " + new Buffer(username + ":" + password).toString("base64");
```
Comment (perry-mitchell) [inline source/auth.js]: I think having the braces around the `new Buffer()` is much nicer. `new Something().blah()` is ugly and misleading.

Comment (perry-mitchell) [inline source/auth.js]: I wonder if there's a prettier setting for this..

Comment (perry-mitchell) [inline source/auth.js]: I guess there isn't.. nevermind.

**Comment by zh99998** [comment]
yes you are the boss, you always right, We Will Follow You Only

**Comment by zh99998** [comment]
ohh it failed..

**Comment by zh99998** [comment]
done.

new Buffer() call is deprecated in node.js now since you don't like it.

File: .editorconfig [hunk 3]
```diff
@@ -7,7 +7,7 @@ indent_style = space
 indent_size = 4
 
 [package.json]
-indent_size = 2
+indent_size = 4
```
Comment (perry-mitchell) [inline .editorconfig]: I just meant for the code, package.json + other existing configs could stay at 2 spaces for simplicity. Many programs that generate them automatically only use 2 spaces (like `npm install` for example). Best leave it like that perhaps

**Comment by perry-mitchell** [comment]
> yes you are the boss, you always right, We Will Follow You Only

Perhaps you could avoid the patronising comments, if you don't mind. I think I've been fair in my requests..

> new Buffer() call is deprecated in node.js now since you don't like it.

Thanks for changing that, looks much better now.

I added only 1 more comment. Otherwise this MR looks very good.

**Comment by zh99998** [comment]
fixed package.json indent size back to 2

**Pull request merged**